#23692: improve build time by avoiding unnecessarry resources filtering

lukasj opened the pull request:
having legal resources in an extra artifact improves the speed of the build by not filtering source tree in every module

Relates to #23692

lukasj committed:
#23692: improve build time by avoiding unnecessarry resources filtering

Signed-off-by: Lukas Jungmann <lukas.jungmann@oracle.com>

lukasj committed:
Hot fix broken build, no idea if/how it is passing without this

Signed-off-by: Lukas Jungmann <lukas.jungmann@oracle.com>

dmatej commented on nucleus/distributions/pom.xml:
```diff
@@ -148,6 +148,7 @@
                         <attach>false</attach>
                         <appendAssemblyId>false</appendAssemblyId>
                         <useProjectArtifact>false</useProjectArtifact>
+                        <ignoreMissingDescriptor>true</ignoreMissingDescriptor>
```
The same property is set to false here. Also I believe false is correct - if the descriptor is missing, the build is broken.

dmatej commented on nucleus/parent/pom.xml:
```diff
@@ -928,11 +930,7 @@
                         <configuration>
                             <resources>
                                 <resource>
-                                    <directory>${maven.multiModuleProjectDirectory}</directory>
```
It worked, because the maven.multiModuleProjectDirectory pointed to the root of the project. But it was rather emergency temporal solution. This is better.

lukasj replied:
there was a discussion about that property, see: https://github.com/eclipse-ee4j/eclipselink/pull/1244#discussion_r693939448

dmatej replied:
It is still somehow nasty, but less nasty than before ;-)

dmatej commented on nucleus/parent/pom.xml:
```diff
@@ -962,6 +960,51 @@
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
groupId is redundant (implicit in both cases)

lukasj replied:
that does not mean one cannot use it ;-)

dmatej commented on appserver/distributions/pom.xml:
```diff
@@ -172,6 +172,7 @@
                         <attach>false</attach>
                         <appendAssemblyId>false</appendAssemblyId>
                         <useProjectArtifact>false</useProjectArtifact>
+                        <ignoreMissingDescriptor>true</ignoreMissingDescriptor>
```
dtto, ignoreMissingDescriptor is already set to false, which I believe is correct

dmatej reviewed (changes requested):
It worked because the assembly plugin is just in plugin management, while you added it to the implicit lifecycle, just without the not inherited common-resources execution (now it executes default execution from plugin management even where it wasn't desired).

It is really interesting, I had to compare both builds :)

dmatej commented:
Funny, I would think that this is Maven's bug - even when I add <inherit>false</inherit> both to plugin and your new execution, it is always executed when it is mentioned just in pluginManagement. Evil :D

lukasj commented:
>Funny, I would think that this is Maven's bug - even when I add false both to plugin and your new execution, it is always executed when it is mentioned just in pluginManagement. Evil :D

No, it is not a bug. It is caused by the fact that the execution id defined in the plugin management is `default-single` - that implies that it gets executed every time. Changing the id to something else, probably with related refactoring, should help as well

dmatej commented:
Without your changes it wasn't executed - and it shouldn't be executed here. Plugin management touches just configured plugins in the lifecycle and there wasn't any such plugin. You added it just to the nucleus-parent and anything from there should not be inherited, but it is. The plugin then takes configuration from the plugin management.

dmatej commented:
Ok, nice, nucleus-common used wrong (meaningless) id, now it starts making sense ... I hope we are on the good path :)

dmatej commented:
Wow. Finally it is not a bug but feature ... maven assembly plugin simply brings default-single by default if there is no execution active, but the plugin is there (but still ... if said inhertited=false, I would expect that even that plugin will not be active). I am curious how far we can get when this will join #23696 

`Total time:  01:36 min (Wall Clock)`

dmatej committed:
Issue #23692 Fixed build

- default-single is brought by maven if we configured plugin execution in pluginManagment
  but not in plugins AND is mentioned in any parent (even if inherited=false)
- so I removed the execution and left it's configuration
- now the default-single is used just in 4 cases:
  atomic, web, glassfish, nucleus-new
- explicit execution is just in cases when we need the assembly sooner (usually
  for the glassfishbuild-maven-plugin)

Signed-off-by: David Matějček <dmatej@seznam.cz>

dmatej commented:
I have created PR for your PR (I didn't want to aggresively push here, just take a look for sure):
https://github.com/lukasj/glassfish-1/pull/1
